src/medImageIO/itkDataImageReaderBase.cpp

```diff
@@ -242,21 +242,79 @@ bool itkDataImageReaderBase::read_image(const QString& path,const char* type)
     TReader->SetImageIO(this->io);
     TReader->SetFileName(path.toAscii().constData());
     TReader->SetUseStreaming(true);
+    TReader->Update();
+
     typename Image::Pointer im = TReader->GetOutput();
     medData->setData(im);
 
-    //  Recover the pixel meaning from the intent_name.
-
     const itk::MetaDataDictionary dict = im->GetMetaDataDictionary();
+
+    //  Recover the pixel meaning from the intent_name.
     std::string PixMeaning;
     if (itk::ExposeMetaData(dict,"intent_name",PixMeaning))
         medData->addMetaData(medAbstractImageData::PixelMeaningMetaData,QString(PixMeaning.c_str()));
 
-    TReader->Update();
+    // Add meta data to the medAbstractData volume
+    QStringList keyList = metaDataKeysToCopyForDerivedData();
+    std::string value;
+    foreach(QString key, keyList)
+    {
+        if (itk::ExposeMetaData(dict, key.toStdString(), value))
+        {
+            medData->setMetaData(key, QString(value.c_str()));
+        }
+    }
 
+    if (itk::ExposeMetaData(dict, "MED_MODALITY", value))
+    {
+        medData->setMetaData(medMetaDataKeys::Modality.key(), QString(value.c_str()));
+    }
+    if (itk::ExposeMetaData(dict, "MED_ORIENTATION", value))
+    {
+        medData->setMetaData(medMetaDataKeys::Orientation.key(), QString(value.c_str()));
+    }
     return true;
 }
 
+QStringList itkDataImageReaderBase::metaDataKeysToCopyForDerivedData()
```

Comment:
technically we are not setting the metadata of derived data in this code. ```metaDataKeysToCopy```is sufficient here